### PR TITLE
added troubleshooting question for self-signed certs

### PR DESCRIPTION
**Pull request by cvasilak**
@matzew @abstractj mind to review?

### CODE REVIEW

**Comment by abstractj** [comment]
@cvasilak @matzew I still don't see the big advantage on it over HTTP only. There's no security on bypassing the SSL verification.

**Comment by cvasilak** [comment]
@abstractj understand your concerns, mainly was to cater for similar issues as the one [opened](https://github.com/aerogear/aerogear-ios-push/pull/46) by the user so if the same question raises again in the future, we can point to this page. I have explicitly commented that is _not_ recommended practice and should be avoided, the user is responsible if s/he follows it. Note also, no changes whatsoever in our library code have been made regarding this.

That said, if you believe we should not do it at all, I am totally fine about it.

File: docs/unifiedpush/aerogear-push-ios/troubleshooting.asciidoc [hunk 1]
```diff
@@ -101,3 +101,50 @@ Since iO8, registerForRemoteNotificationTypes has been deprecated in favour of r
...
+
+[source,c]
+----
+Domain=NSURLErrorDomain Code=-1202 "The certificate for this server is invalid. You might be connecting to a server that is pretending to be “example.com” which could put your confidential information at risk." UserInfo=0x14a730 {NSErrorFailingURLStringKey=https://example.com/, NSLocalizedRecoverySuggestion=Would you like to connect to the server anyway?, ..}
+----
+
+*Answer*
+
+By default, the underlying networking foundation rejects connections to server's that use a self-signed certificate and you need to take extra measures in order to make it work. Although the following snippets of code can bypass the built-in security check, they are not recommended unless you are working in a development sandbox and you need a quick-and-dirty solution for internal testing. You can find more information on self-signed certificates on link:https://developer.apple.com/library/ios/technotes/tn2232/_index.html#//apple_ref/doc/uid/DTS40012884-CH1-SECSELFSIGNEDCERTS[Apple technote paper] as well in link:https://github.com/aerogear/aerogear-ios-push/pull/46#issuecomment-75815787[this page here] that summarizes possible solutions when you take your app in production.
```
Comment (matzew) [inline docs/unifiedpush/aerogear-push-ios/troubleshooting.asciidoc]: Let's make the _**not recommended**_ bold and italic :-)

**Comment by matzew** [comment]
@abstractj @cvasilak without this work-around, how would an iOS device actually connect a server, running a self-signed certificate? For instance our Docker image is using such a self-signed certificate and I think for testing/development this "solution" or work-around (or hack) is needed, right? 

If that is not needed, yeah, let's not add this document.

**Comment by abstractj** [comment]
@matzew Docker image has a self-signed certificate, because Keycloak and Push server requires one. It **does not** add any sense of security between client/server. The solution for testing environments is: use **HTTP** or buy a certificate.

@cvasilak totally fine with your change, if you think that can be helpful.

**Comment by abstractj** [comment]
@cvasilak landed

**Pull request merged**